oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -430,6 +428,7 @@ public GoogleCredentials createScoped(Collection<String> scopes) {
         .setHttpTransportFactory(this.transportFactory)
         .setQuotaProjectId(this.quotaProjectId)
         .setIamEndpointOverride(this.iamEndpointOverride)
+        .setAccessToken(null)
```

Comment:
Why do we have to explicitly set it to null?
Also thanks to the newly introduced constructor below `protected Builder(ImpersonatedCredentials credentials)`, I think this can be simplified to 
```
toBuilder()
        .setScopes(new ArrayList<>(scopes))
        .build()
```

Reply:
> Why do we have to explicitly set it to null?

This is invoking the parent class builder which sets: 
https://github.com/googleapis/google-auth-library-java/blob/89201558db913d9a71b3acccbab8eb0045ada6de/oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java#L648
We have #1428, #1489 fixing similar situations in other credential classes.

You are right about the other setters.

Reply:
> This is invoking the parent class builder
Ah, makes sense, thanks! Basically we want to invalidate existing tokens when creating a new type of credential.